frostsnap_core/src/device.rs

```diff
@@ -455,17 +472,19 @@ pub enum SignerState {
     Registered,
     KeyGen {
         scalar_poly: Vec<Scalar>,
-        devices: BTreeSet<DeviceId>,
+        devices: BTreeMap<DeviceId, Scalar<Public, NonZero>>,
         threshold: usize,
         aux_rand: [u8; 32],
     },
     AwaitingSignAck {
         key: FrostsnapKey,
+        devices: BTreeMap<DeviceId, Scalar<Public, NonZero>>,
         sign_task: SignTask,
         nonces: BTreeMap<DeviceId, (Vec<Nonce>, usize, usize)>,
```

Comment:
Do we need to know the device id's of the other devices after key generation? I think I just need to know my index.
We shouldn't have the `DeviceId` in the nonces either, it should just be the signer index.

Same is true for the `FrostKey` state. Just need to know my index.

Reply:
We're using sets of `DeviceId` in each `CoordinatorToDevice` variant to get `default_destinations()`, so i added `targets` on `CoordinatorToDevice::SignRequest`.